### PR TITLE
Update ModuleDict doc about order

### PR DESCRIPTION
**Pull request by ssnl**

### CODE REVIEW

File: torch/nn/modules/container.py [hunk 1]
```diff
@@ -207,8 +207,21 @@ def extend(self, modules):
...
+    :class:`~torch.nn.Module` methods.
+
+    :class:`~torch.nn.ModuleDict` is an **ordered** dictionary that respects
+
+    * the order of insertion, and
+
+    * in :meth:`~torch.nn.ModuleDict.update`, the order of the merged ``OrderedDict``
+      or another :class:`~torch.nn.ModuleDict` (the argument to :meth:`~torch.nn.ModuleDict.update`).
+      
+      Note that :meth:`~torch.nn.ModuleDict.update` with other unordered mapping
```
Comment (vishwakftw) [inline torch/nn/modules/container.py]: Is the (mis)alignment intended?

File: torch/nn/modules/container.py [hunk 2]
```diff
@@ -207,8 +207,21 @@ def extend(self, modules):
...
+    but modules it contains are properly registered, and will be visible by all 
+    :class:`~torch.nn.Module` methods.
+
+    :class:`~torch.nn.ModuleDict` is an **ordered** dictionary that respects
+
+    * the order of insertion, and
+
+    * in :meth:`~torch.nn.ModuleDict.update`, the order of the merged ``OrderedDict``
+      or another :class:`~torch.nn.ModuleDict` (the argument to :meth:`~torch.nn.ModuleDict.update`).
+
```
Comment (vishwakftw) [inline torch/nn/modules/container.py]: There seems to be a newline here.

Comment (SsnL) [reply torch/nn/modules/container.py]: thanks! fixed!

File: torch/nn/modules/container.py (hunk 1 above)
Comment (SsnL) [reply torch/nn/modules/container.py]: thanks! fixed!

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**